### PR TITLE
GNS3 and dynamips updates

### PR DESCRIPTION
**Pull request by tsndqst**
Upgrade dynamips and cross-compile when binfmt is enabled.
Upgrade gns3-server and enable more archictures since x86_64 busybox is no longer packaged with gns3-server.
Upgrade gns3-gui.

#### Testing the changes
- I tested the changes in this PR: **YES**
- I tested the `Ethernet Switch` on aarch64 which uses the dynamips hypervisor
- I tested Alpine Linux docker container on aarch64 as well.

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-GLIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64
  - aarch64 (crossbuild)

### CODE REVIEW

File: srcpkgs/gns3-server/template [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Template file for 'gns3-server'
 pkgname=gns3-server
-version=2.2.35
+version=2.2.36
 revision=1
 build_style=python3-module
-hostmakedepends="python3-setuptools"
+hostmakedepends="python3-setuptools busybox-static"
```
[the remark this answers is not in the document]
Comment (tsndqst) [reply srcpkgs/gns3-server/template:6]: I think you're right @CameronNemo.   https://github.com/GNS3/gns3-server/issues/1890 includes the change to use a stock busybox to inject into Docker images.  I'll test it out on arm64 and see if I can get it working.

**Comment by tsndqst** [comment]
I updated the templates to cross compile dynamips and gns3-server.  I tested it locally with aarch64 and they both worked.  I'll have to dig into the CI failures a bit when I have some time.

**Comment by tsndqst** [comment]
I was able to reproduce the problem on one of my x86_64 machines but I can't figure out what the difference is between the machine that can cross compile dynamips and the one that can't.  The host xbps-src runs on shouldn't affect the outcome but something is different between the two. Got any hints about how to track down the difference?

**Comment by tsndqst** [comment]
> Off hand I would say that zapping masterdir and clearing out any stale binpkgs would be good troubleshooting tactic.

I tried it on a fresh clone of `void-packages` and it still worked fine.  I'll keep digging.

**Comment by tsndqst** [comment]
@CameronNemo can you re-review this?  GNS3 was updated since I created this PR so I bumped the version since the last time you looked at it.

File: srcpkgs/dynamips/template [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-# Dynamips does processor instruction translation.  This is already
-# painfully slow on powerful x86 hardware and is unlikely to work
-# reliably on arm processors.
-#
-# So restrict to x86_64* and i686*
+_BINFMT_PATH=/proc/sys/fs/binfmt_misc/qemu-${XBPS_TARGET_QEMU_MACHINE}-static
+
+if ! grep -q enabled ${_BINFMT_PATH} 2>/dev/null; then
+	nocross="binfmt support not enabled"
```
[the remark this answers is not in the document]
Comment (tsndqst) [reply srcpkgs/dynamips/template:19]: binfmt support is not available in Github CI so we can't cross compile in Github but if binfmt is supported locally we can cross compile.

[the remark this answers is not in the document]
Comment (tsndqst) [reply srcpkgs/dynamips/template:19]: Good idea.

**Comment by tsndqst** [comment]
I pushed an update since GNS3 2.2.39 was released.

**Comment by tsndqst** [comment]
I pushed an update since GNS3 2.2.40 was released.

**Comment by SpidFightFR** [comment]
Heya for the build fails, the answer lies here: https://github.com/GNS3/gns3-server/issues/2305#issuecomment-1784473048

**Comment by Vaelatern** [comment]
@tsndqst I've got a PR open: #47331

I'd rather merge your PR.

Please update to 2.2.44, drop the `.patch` file... I'm not sure `makedepends="${hostmakedepends}` is necessary, the `hostmakedepends` bit. Please either justify the extra python pkg or not.

Also please do not change PATH to prefix with the cross dir. GNS3's build has an environment variable, I believe `$BUSYBOX_PATH` or similar, which it checks before looking on `PATH` for a `busybox.static`. Set that instead.

`python3-platformdirs` is a `depends` not a `checkdepends`. See my PR #47331.

With these changes, I'll merge this PR.

I'm currently using GNS3 on my x86_64-musl machine.

**Comment by tsndqst** [comment]
Thanks for reminding me to look at this again @Vaelatern.  I was waiting for the 2.2.44 release then got distracted by the aiohttp 3.9 release that supports Python 3.12.  I  haven't used GNS3 recently but perhaps it works fine on 3.12 without the latest aiohttp library.

Anyway, I have integrated your suggestions into my local branch and I'm trying to figure out how to manage the busybox binary but I'm not having much luck.  The BUSYBOX_PATH variable you found is a Python variable.  I tried setting it via an environment variable so I can change it based on XBPS variables during build.  I haven't found a combination that works yet.  

Also, it turns out `setup.py` uses that BUSYBOX_PATH variable as a destination.  It is where GNS3 looks for binaries to copy into docker images.  It uses the `shutil.which()` to find the appropriate binary. 

I ended up copying the static busybox file into the location where the GNS3 setup expects to find it.  I think that is cleaner than what I had before.

I also tested it on my aarch64 box and docker images contain the aarch64 static busybox binary.

**Comment by Vaelatern** [comment]
> Also, it turns out setup.py uses that BUSYBOX_PATH variable as a destination. It is where GNS3 looks for binaries to copy into docker images. It uses the shutil.which() to find the appropriate binary.

Good catch!

**Pull request merged**